core/src/value/person_name.rs

```diff
@@ -0,0 +1,326 @@
+use crate::dicom_value;
+use crate::value::PrimitiveValue;
+use snafu::{OptionExt, Snafu};
+use std::fmt::{Display, Formatter};
+
+#[derive(Debug, Snafu)]
+pub enum Error {
+    #[snafu(display("PatientName cannot start with '^'"))]
+    LeadingWhitespace {},
+    #[snafu(display("Family name is mandatory"))]
+    NoFamily {},
+    #[snafu(display("Family name is empty"))]
+    EmptyFamily {},
+    #[snafu(display("Given name is mandatory"))]
+    NoGiven {},
+    #[snafu(display("Given name is epmty"))]
+    EmptyGiven {},
+}
+pub type Result<T, E = Error> = std::result::Result<T, E>;
+
+/// Represents a Dicom PersonName.
+/// Contains family, given, middle name, prefix and suffix.
+/// For practical purposes, family and given name are mandatory components, not optional as per standard.
+/// The Dicom string representation is split by "^" separator into its respective components.
+/// Does not support component group delimiter "=", ideographic and phonetic characters.
+#[derive(Debug, Clone, PartialEq)]
+pub struct PersonName<'a> {
+    prefix: Option<&'a str>,
+    family: &'a str,
+    middle: Option<&'a str>,
+    given: &'a str,
+    suffix: Option<&'a str>,
+}
+
+impl<'a> From<PersonName<'a>> for PrimitiveValue {
+    fn from(p: PersonName) -> Self {
+        dicom_value!(Str, p.to_dicom_string())
+    }
+}
+
+macro_rules! write_if_some {
+    ($formater: expr, $option: expr,  $format: expr) => {
+        if let Some(opt) = $option {
+            write!($formater, $format, opt)?
+        }
+    };
+}
+
+impl Display for PersonName<'_> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write_if_some!(f, self.prefix, "{} ");
+        write!(f, "{} ", self.family)?;
+        write_if_some!(f, self.middle, "{} ");
+        write!(f, "{}", self.given)?;
+        write_if_some!(f, self.suffix, " {}");
+        Ok(())
+    }
+}
+
+impl<'a> PersonName<'a> {
+    pub fn new(
+        prefix: Option<&'a str>,
+        given: &'a str,
+        family: &'a str,
+        middle: Option<&'a str>,
+        suffix: Option<&'a str>,
+    ) -> Self {
+        Self {
+            prefix,
+            family,
+            middle,
+            given,
+            suffix,
+        }
+    }
+    pub fn to_dicom_string(&self) -> String {
+        let mut name = format!("{}^{}", self.family, self.given);
+        match self.middle {
+            Some(middle) => name.push_str(&format!("^{}", middle)),
```

Comment:
Clippy is complaining about this, probably because by calling `name.push('^')` and `name.push_str(middle)`, we save a string allocation. Same for the other cases below.